[REF] Inventory and MRP: Update TOC tree

hojo-odoo opened the pull request:
This PR overhauls the "Inventory & MRP" TOC tree, including:

- Moving the Barcode folder out of the Inventory folder and into the main Inventory & MRP folder
- Moving the Maintenance and Quality folders out of the Manufacturing folder and into the main Inventory & MRP folder
- Adding app descriptions and see alsos for the main Barcode, Maintenance, and Quality app pages

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/documentation/pull/5591).

jcs-odoo commented on content/applications/inventory_and_mrp/quality.rst:
```diff
@@ -0,0 +1,20 @@
+.. toctree::
+   :titlesonly:
+
+   quality/quality_management
```
I think this category is useless, and you could list `quality_control_points` and `quality_alerts` directly here, wdyt?

hojo-odoo replied:
I will be fleshing out the Quality scope with more docs over the coming months. They will need to be split across multiple sub-folders. I've already created the "Quality management" sub-folder for the control points and alerts docs in anticipation of that. I could remove the sub-folder for now, but it's going to be added back in shortly.

AntoineVDV commented on content/applications/inventory_and_mrp.rst:
```diff
@@ -10,3 +10,6 @@ Inventory & MRP
     inventory_and_mrp/inventory
     inventory_and_mrp/manufacturing
     inventory_and_mrp/purchase
+    inventory_and_mrp/barcode
+    inventory_and_mrp/quality
+    inventory_and_mrp/maintenance
```
Done. I kept the original teams as codeowners, but it can be changed after.

Felicious commented on content/applications/productivity/iot/devices/camera.rst:
```diff
@@ -55,8 +55,8 @@ is reached during the manufacturing process, the database prompts the operator t
 .. seealso::
-   - :doc:`../../../inventory_and_mrp/manufacturing/quality_control/quality_control_points`
-   - :doc:`../../../inventory_and_mrp/manufacturing/quality_control/quality_alerts`
+   - :doc:`../../../inventory_and_mrp/quality/quality_management/quality_control_points`
+   - :doc:`../../../inventory_and_mrp/quality/quality_management/quality_alerts`
```
I was told by Sam that when linking to docs, we want to use absolute paths, like the `/applications/....` format that we used in this [Documentation guideline](https://www.notion.so/Documentation-Guidelines-78bc0a486181431ebab8d4145064eafa?pvs=4#d078e791906f4f87aca3ab6b8c6bce05). Not to mansplain to you, but what's used here is a [relative path](https://www.codingrooms.com/blog/file-paths#:~:text=A%20relative%20path%20describes%20the,regularly%20use%20relative%20file%20paths.)! 
```suggestion
   - :doc:`Quality control points
     </applications/inventory_and_mrp/quality/quality_management/quality_control_points>`
   - :doc:`Quality alerts
     </applications/inventory_and_mrp/quality/quality_management/quality_alerts>`
```
Also in my suggestion, I added a line because the whole thing would've exceeded 100 chars for the line!

Felicious commented on content/applications/sales/point_of_sale/shop/barcode.rst:
```diff
@@ -18,8 +18,8 @@ To use a barcode scanner, you must enable the feature in the Inventory app. Go t
 .. seealso::
-   - :doc:`Set up a barcode scanner<../../../inventory_and_mrp/inventory/barcode/setup/hardware>`
-   - :doc:`Activate barcode scanners<../../../inventory_and_mrp/inventory/barcode/setup/software>`
+   - :doc:`Set up a barcode scanner<../../../inventory_and_mrp/barcode/setup/hardware>`
+   - :doc:`Activate barcode scanners<../../../inventory_and_mrp/barcode/setup/software>`
```
```suggestion
   - :doc:`Set up a barcode scanner </applications//inventory_and_mrp/barcode/setup/hardware>`
   - :doc:`Activate barcode scanners </applications/inventory_and_mrp/barcode/setup/software>`
```

Felicious commented on content/applications/productivity/iot/devices/printer.rst:
```diff
@@ -67,8 +67,8 @@ specific product.
 .. seealso::
-   - :doc:`../../../inventory_and_mrp/manufacturing/quality_control/quality_control_points`
-   - :doc:`../../../inventory_and_mrp/manufacturing/quality_control/quality_alerts`
+   - :doc:`../../../inventory_and_mrp/quality/quality_management/quality_control_points`
+   - :doc:`../../../inventory_and_mrp/quality/quality_management/quality_alerts`
```
Same here! just doing this so it's easier for you (:
```suggestion
   :doc:`Quality control points
      </applications/inventory_and_mrp/quality/quality_management/quality_control_points>`
   - :doc:`Quality alerts
     </applications/inventory_and_mrp/quality/quality_management/quality_alerts>`
```

Felicious commented on content/applications/inventory_and_mrp/quality.rst:
```diff
@@ -0,0 +1,20 @@
+when issues arise.
+
+.. seealso::
+   - `Odoo Tutorials: Managing Quality Checks <https://www.odoo.com/slides/slide/managing-quality-checks-708>`_
```
Exceed 100 chars
```suggestion
      - `Odoo Tutorials: Managing Quality Checks
     <https://www.odoo.com/slides/slide/managing-quality-checks-708>`_
```

samueljlieber replied:
Can fix this line over 100 chars too :)
```suggestion
   - `Odoo Tutorials: Managing Quality Checks
     <https://www.odoo.com/slides/slide/managing-quality-checks-708>`_
```

Felicious commented on content/applications/inventory_and_mrp/maintenance.rst:
```diff
@@ -0,0 +1,19 @@
+requirements.
+
+.. seealso::
+   - `Odoo Tutorials: Managing Equipment Maintenance <https://www.odoo.com/slides/slide/managing-equipment-maintenance-709>`_
```
Exceed 100 chars
```suggestion
   - `Odoo Tutorials: Managing Equipment Maintenance
     <https://www.odoo.com/slides/slide/managing-equipment-maintenance-709>`_
```

samueljlieber replied:
May as well fix this line over 100 chars while were at it!
```suggestion
   - `Odoo Tutorials: Managing Equipment Maintenance
     <https://www.odoo.com/slides/slide/managing-equipment-maintenance-709>`_
```

Felicious commented on content/applications/inventory_and_mrp/barcode.rst:
```diff
@@ -0,0 +1,22 @@
+Barcode
+=======
+
+**Odoo Barcode** allows users to assign barcodes to individual products and product categories, and
+track inventory movements using those barcodes. By connecting a barcode scanner, certain inventory
+processes can be triggered by scanning barcodes.
```
This is great, and nothing about this is wrong; I just see the barcode app from a different angle. Feel free to reject/modify my description of the barcode app!

 ```suggestion
**Odoo Barcode** allows users to streamline warehouse receiving, picking, delivering, and performing
inventory adjustments on products and product categories. Reduce errors and log warehouse operations
with a single scan. Use with **Odoo Inventory** to assign barcodes to individual products and
product categories, and track inventory movements across the supply chain.
```

Felicious reviewed (commented):
Amazing job on this massive PR that affects the entire materials scope! I can tell you're very meticulous and comprehensive with your work by how there are no failures to CI checks :smiley: 

I just had some suggestions on changing relative paths to absolute, and caught some lines exceeding 100 characters; all very minor suggestions. Again, great job, John, and I'm really happy you reorganized everything for us!

hojo-odoo commented on content/applications/inventory_and_mrp/maintenance.rst:
```diff
@@ -0,0 +1,19 @@
+.. toctree::
+   :titlesonly:
+
+   maintenance/maintenance_management
```
Like the Quality sub-folder, I've put the "Add new equipment" doc into a management sub-folder in anticipation of the Maintenance scope being fleshed out with more documentation.

StraubCreative commented:
@samueljlieber can you thoroughly review this please? Bit of a big lift but very exiting! 🚀

samueljlieber reviewed (changes requested):
Hi @hojo-odoo! Awesome work with this restructure 💪 I have some technical corrections and suggestions for you:

1. Missing redirects for all file moves:
```txt
# applications/inventory_and_mrp

applications/inventory_and_mrp/inventory/barcode.rst applications/inventory_and_mrp/barcode.rst                                                                                  # /inventory_and_mrp/inventory/barcode/* -> /inventory_and_mrp/barcode/*
applications/inventory_and_mrp/inventory/barcode/operations.rst applications/inventory_and_mrp/barcode/operations.rst                                                            # /inventory_and_mrp/inventory/barcode/* -> /inventory_and_mrp/barcode/*
applications/inventory_and_mrp/inventory/barcode/operations/adjustments.rst applications/inventory_and_mrp/barcode/operations/adjustments.rst                                    # /inventory_and_mrp/inventory/barcode/* -> /inventory_and_mrp/barcode/*
applications/inventory_and_mrp/inventory/barcode/operations/barcode_nomenclature.rst applications/inventory_and_mrp/barcode/operations/barcode_nomenclature.rst                  # /inventory_and_mrp/inventory/barcode/* -> /inventory_and_mrp/barcode/*
applications/inventory_and_mrp/inventory/barcode/operations/gs1_nomenclature.rst applications/inventory_and_mrp/barcode/operations/gs1_nomenclature.rst                          # /inventory_and_mrp/inventory/barcode/* -> /inventory_and_mrp/barcode/*
applications/inventory_and_mrp/inventory/barcode/operations/internal.rst applications/inventory_and_mrp/barcode/operations/internal.rst                                          # /inventory_and_mrp/inventory/barcode/* -> /inventory_and_mrp/barcode/*
applications/inventory_and_mrp/inventory/barcode/operations/transfers_scratch.rst applications/inventory_and_mrp/barcode/operations/transfers_scratch.rst                        # /inventory_and_mrp/inventory/barcode/* -> /inventory_and_mrp/barcode/*
applications/inventory_and_mrp/inventory/barcode/setup.rst applications/inventory_and_mrp/barcode/setup.rst                                                                      # /inventory_and_mrp/inventory/barcode/* -> /inventory_and_mrp/barcode/*
applications/inventory_and_mrp/inventory/barcode/setup/hardware.rst applications/inventory_and_mrp/barcode/setup/hardware.rst                                                    # /inventory_and_mrp/inventory/barcode/* -> /inventory_and_mrp/barcode/*
applications/inventory_and_mrp/inventory/barcode/setup/software.rst applications/inventory_and_mrp/barcode/setup/software.rst                                                    # /inventory_and_mrp/inventory/barcode/* -> /inventory_and_mrp/barcode/*
applications/inventory_and_mrp/manufacturing/maintenance.rst applications/inventory_and_mrp/maintenance/maintenance_management.rst                                               # /inventory_and_mrp/manufacturing/maintenance/* -> /inventory_and_mrp/maintenance/*
applications/inventory_and_mrp/manufacturing/maintenance/add_new_equipment.rst applications/inventory_and_mrp/maintenance/maintenance_management/add_new_equipment.rst           # /inventory_and_mrp/manufacturing/maintenance/* -> /inventory_and_mrp/maintenance/*
applications/inventory_and_mrp/manufacturing/quality_control.rst applications/inventory_and_mrp/quality.rst                                                                      # /inventory_and_mrp/manufacturing/quality_control/* -> /inventory_and_mrp/quality/*
applications/inventory_and_mrp/manufacturing/quality_control/quality_alerts.rst applications/inventory_and_mrp/quality/quality_management/quality_alerts.rst                     # /inventory_and_mrp/manufacturing/quality_control/* -> /inventory_and_mrp/quality/*
applications/inventory_and_mrp/manufacturing/quality_control/quality_control_points.rst applications/inventory_and_mrp/quality/quality_management/quality_control_points.rst     # /inventory_and_mrp/manufacturing/quality_control/* -> /inventory_and_mrp/quality/*
```

2. The main purpose of this PR is restructuring the `inventory_and_mrp/` directory, this falls under the `[REF]` (refactoring/restructuring) [commit tag](https://github.com/odoo/documentation/blob/16.0/commit_template.txt). Please update the commit accordingly using `git commit --amend` or during your next squash 🙂


<details><summary>Helpful: I mapped out the structure change so its visually clear what this PR restructures</summary>
<p>

🟡 Moved folder
🟢 New file

<h2>Current structure</h2>
<code>inventory_and_mrp/
├── inventory/
│   ├── barcode/
│   │   ├── operations/
│   │   │   ├── adjustments
│   │   │   ├── barcode_nomenclature
│   │   │   ├── gs1_nomenclature
│   │   │   ├── internal
│   │   │   └── transfers_scratch
│   │   └── setup/
│   │       ├── hardware
│   │       └── software
│   ├── management
│   ├── routes
│   └── shipping
├── manufacturing/
│   ├── maintenance/
│   │   └── add_new_equipment
│   ├── management
│   └── quality_control/
│       ├── quality_alerts
│       └── quality_control_points
└── purchase</code>

<h2>New structure</h2>
<code>inventory_and_mrp/
├── 🟡 barcode/
│   ├── operations/
│   │   ├── adjustments
│   │   ├── barcode_nomenclature
│   │   ├── gs1_nomenclature
│   │   ├── internal
│   │   └── transfers_scratch
│   └── setup/
│       ├── hardware
│       └── software
├── inventory/
│   ├── management
│   ├── routes
│   └── shipping
├── 🟡 maintenance/
│   └── 🟢 maintenance_management/
│       └── add_new_equipment
├── manufacturing/
│   └── management
├── purchase
└── 🟡 quality/
    └── 🟢 quality_management/
        ├── quality_alerts
        └── quality_control_points</code>
</p>
</details>

---

Overall, really great work. Please tag me again for another look once my suggestions are addressed!

Thank you!

hojo-odoo commented:
Hey @samueljlieber,

This is ready for another look. I've implemented all of your requested changes.

Thanks,
hojo

samueljlieber reviewed (approved):
Hi @hojo-odoo thank you for implementing my suggestions, this PR is looking great! I am approving now, on to you @StraubCreative 🙂

p.s. I'm updating the PR title to match the commit tag

StraubCreative reviewed (commented):
Hi @hojo-odoo 

It seems, technically, that the toctree structure + RST is sound. I trust all of the 301s are added as well.

From a readability standpoint:
- I see inconsistencies with title vs. sentence cased section headings sprinkled throughout the MRP scope. They should *all* be sentence case. 
- the generic "[app] management" section titles aren't very informative or compelling. **Quality management**, for example, could be **Controls and alerts** or something more descriptive. 
    - It's the same issue for the other sections like "Manufacturing management," where there could very easily be dedicated sections just for BoMs, just for Work Orders, etc.
    - The specificity in section headings educates readers what's inside the app, what's important or worth paying attention to, and in what order concepts should be consumed. However, vague, catch-all umbrella headings like "management" do not aid the reader in any of those ways.

hojo-odoo commented:
Hey @StraubCreative,

I've implemented all of your comments. All of the section titles are sentence case now and I've renamed the [app] management sections.

Thanks,
hojo

StraubCreative reviewed (approved):
> I've implemented all of your comments. All of the section titles are sentence case now and I've renamed the [app] management sections.

These titles make more sense to me now :) 
Can keep dialing those in, as needed, as the sections are built out more.

@jcs-odoo @AntoineVDV @Xavier-Do any feedback for us before we try to merge?

The TL;DR is we wanted to make Barcode, Maintenance, and Quality their own parent sections inside of the Inv/MRP scope and then rapidly build them out going into Q4 through Q1 2024. The sections seem skim for now, but it's only temporary 😅

StraubCreative commented:
Fresh rebase on [5055d3b](https://github.com/odoo/documentation/pull/5591/commits/5055d3b5f672389e7a0f9e1f77c6657a9a7cb152). 

In order to make further content additions we have planned into the newly created lean directories, we need to merge in this org change. Any remaining organizational improvements not resolved in this PR can be addressed later— just make a Task and tag us with your feedback, we will address asap! 🙂 

All checks pass ✅ 

@robodoo r+
